build-tools/automation/yaml-templates/apk-instrumentation.yaml

```diff
@@ -19,20 +20,20 @@ steps:
       /t:AcquireAndroidTarget,SignAndroidPackage,DeployTest${{ parameters.packageType }}s,RunTestApks,UndeployTestApks,RenameApkTestCases,ReportComponentFailures
       /bl:$(System.DefaultWorkingDirectory)/bin/Test${{ parameters.configuration }}/run${{ parameters.testName }}.binlog
       ${{ parameters.extraBuildArgs }}
-  condition: succeededOrFailed()
+  condition: ${{ parameters.condition }}
 
 - script: |
     SOURCE=$(System.DefaultWorkingDirectory)/bin/Test${{ parameters.configuration }}/${{ parameters.artifactName }}
     DEST=$(Build.ArtifactStagingDirectory)/${{ parameters.artifactFolder }}/
     mkdir "$DEST"
     cp "$SOURCE" "$DEST"
   displayName: copy apk/aab
-  condition: succeededOrFailed()
+  condition: ${{ parameters.condition }}
 
 - task: PublishTestResults@2
   displayName: publish ${{ parameters.testName }} results
   inputs:
     testResultsFormat: ${{ parameters.testResultsFormat }}
     testResultsFiles: ${{ parameters.testResultsFiles }}
     testRunTitle: ${{ parameters.testName }}
-  condition: and(succeededOrFailed(), ne('${{ parameters.testResultsFiles }}', ''))
+  condition: ${{ parameters.condition }}
```

Comment:
The latest attempt had a few errors in places where we aren't passing this parameter:
> `Error: Input required: testResultsFiles`.

Looks like we'll need to bring this check back. Perhaps the following will work?

```yaml
and(${{ parameters.condition }}, ne('${{ parameters.testResultsFiles }}', ''))
```